integration-tests/src/test/java/org/apache/fineract/integrationtests/LoanInterestPauseApiTest.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.integrationtests;
+
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.builder.ResponseSpecBuilder;
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+import io.restassured.specification.ResponseSpecification;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.fineract.integrationtests.common.Utils;
+import org.apache.fineract.integrationtests.common.loans.InterestPauseHelper;
+import org.apache.fineract.integrationtests.common.loans.LoanTestLifecycleExtension;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@Slf4j
+@ExtendWith({ LoanTestLifecycleExtension.class })
+public class LoanInterestPauseApiTest extends BaseLoanIntegrationTest {
+
+    private static RequestSpecification REQUEST_SPEC;
+    private static ResponseSpecification RESPONSE_SPEC;
+    private static ResponseSpecification RESPONSE_SPEC_403;
+    private static InterestPauseHelper INTEREST_PAUSE_HELPER;
+
+    @BeforeAll
+    public static void setupTests() {
+        Utils.initializeRESTAssured();
+        REQUEST_SPEC = new RequestSpecBuilder().setContentType(ContentType.JSON).build();
+        REQUEST_SPEC.header("Authorization", "Basic " + Utils.loginIntoServerAndGetBase64EncodedAuthenticationKey());
+        RESPONSE_SPEC = new ResponseSpecBuilder().expectStatusCode(200).build();
+
+        RESPONSE_SPEC_403 = new ResponseSpecBuilder().expectStatusCode(403).build();
+
+        INTEREST_PAUSE_HELPER = new InterestPauseHelper(REQUEST_SPEC, RESPONSE_SPEC);
+    }
+
+    @Test
```

Comment:
Missing test cases:
- Loan does not exist
- Loan id / external id was not provided: `null` or `empty string`
- Fetch interest pauses for a loan before and after at least 1 was created.
- Incorrect from date was provided
- Incorrect till date was provided
- Not matching date format was provided

Reply:
Added 12 tests